Src/BootCamp.Chapter/BinaryConverter.cs

```diff
@@ -8,12 +8,50 @@ public static class BinaryConverter
     {
         public static long ToInteger(string binary)
         {
-            return 0;
+            if (String.IsNullOrEmpty(binary))
+            {
+                return 0;
+            }
+            try
+            {
+                int integerNumber = Convert.ToInt32(binary, 2);
+                return integerNumber;
+            }
+            catch
+            {
+                throw new InvalidBinaryNumberException(binary);
+            }
+        }
+
+        private static bool IsBinary(string binary)
+        {
+            foreach (var item in binary)
+            {
+                if (item != '0' || item != '1')
+                {
+                    return false;
+                }
+            }
+            return true;
         }
 
         public static string ToBinary(long number)
         {
-            return "";
+            Stack<int> myBinaryStack = new Stack<int>();
```

Comment:
You don't need a stack here- you can simply append the number to string right away.

Reply:
Changed, but my way can be little to mess :/